crdt/rlist_property_test.go

```diff
@@ -0,0 +1,66 @@
+package crdt_test
+
+import (
+	"testing"
+
+	"github.com/brunokim/causal-tree/crdt"
+	"pgregory.net/rapid"
+)
+
+// Model an RList as a slice of chars, subject to insertions and deletions
+// at random positions with InsertCharAt and DeleteCharAt.
+//
+// We don't model the more primitive operations InsertChar, DeleteChar and SetCursor
+// because it's complicated to model where the cursor ends up after a deletion. The
+// cursor moves to the deleted atom's cause, or its first non-deleted ancestor,
+// which may not be the char next to it.
+//
+// TODO: perhaps this is a sign that the cursor should be more predictable...?
+type stateMachine struct {
+	l     *crdt.RList
+	chars []rune
+}
+
+func (m *stateMachine) Init(t *rapid.T) {
```

Comment:
Argument t *rapid.T is not used by the function. Is it necessary?

Reply:
Yes, because the signature is used to identify and call this method via reflection: https://pkg.go.dev/pgregory.net/rapid#StateMachine